avro4s-core/src/main/scala/com/sksamuel/avro4s/AvroInputStream.scala

```diff
@@ -15,11 +15,11 @@ trait AvroInputStream[T] {
   def tryIterator(): Iterator[Try[T]]
 }
 
-class AvroBinaryInputStream[T](in: InputStream)
+class AvroBinaryInputStream[T](in: InputStream, writerSchema: org.apache.avro.Schema = null)
```

Comment:
I actually start by writing this version of the api that provide the schema via implicit. but than I realize, that if the class change over time I just lost the writer schema :)